docs-src/concepts/what-is-a-logger.md

```diff
@@ -0,0 +1,33 @@
+---
+id: what-is-a-logger
+title: What is a Logger?
+sidebar_label: Logger
+description: A Logger provides an API containing the logic for recording runtime behavior.
+tags:
+  - term
+  - explanation
+  - logger
+  - api
+  - sdks
+---
+
+A logger provides an API containing the logic needed to record details about the application behavior at runtime.
+The output posts timestamps alongside the logged values, making it easier to evaluate the state and flow of [Workflows](/concepts/what-is-a-workflow) and [Activities](/concepts/what-is-an-activity).
+
+[Temporal SDKs](/concepts/what-is-a-temporal-sdk) provide both a logging API and a built-in logger implementation.
+The logging implementation provided by the SDK emits messages to the standard output of the [Worker](/concepts/what-is-a-worker) executing the code.
+The Temporal logging API automatically injects helpful information (such as [Workflow ID](/concepts/what-is-a-workflow-id)) about the execution into the output.
+In addition, the logging API is replay-aware, preventing the duplication of log messages if History Replay is used to restore the app after a crash.
+
+Although the output is timestamped and given a log level (similar to other logging APIs out there), the SDK does not support output customization.
```

Comment:
> timestamped

This is SDK specific

> the SDK does not support output customization

This is not true. Most logging interfaces let you implement whatever approach you want. And ones using built-in logging frameworks (Java slf4j, Python logging module, .NET logging extension, etc) all support _lots_ of output customization. SDK just uses these like any other library would, there's nothing Temporal specific here in many cases (except for the extra properties we add and the skip-during-replay we offer).

Reply:
> I filed the original issue because a user was confused about where the output from the Workflow and Activity loggers in the Go SDK went and how to change its format and/or destination.

But this doc we're reviewing is not for Go SDK

> AFAIK, there is no way for a developer to customize the format or output destination unless they write their own implementation (or an adapter class for an existing third-party package like zap), yet I was unable to find that documented anywhere.

Yes, it depends on the implementation they have. We provide an interface, how that interface is implemented is up to users. Formatting and output destination are but a couple of the many options one may choose to think about when implementing. We should move zap into contrib (https://github.com/temporalio/sdk-go/issues/704) as well as the upcoming `slog` (https://github.com/temporalio/sdk-go/issues/1158). But this PR doesn't help you solve your original problem of how to customize format or output destination.

> For an SDK like Java, which appears to use SLF4J, that might just be a couple of sentences that say

Right, see https://docs.temporal.io/dev-guide/java/observability#logging. That can be expanded/improved.

> Documentation for other SDKs could go into more detail, where necessary. To be clear, I don't think we need to cover how to configure a third-party logging package, but IMHO we should at least identify which one is being used by the SDK's logger (where applicable) so that people can find the docs provided by that project.

:+1:

Reply:
> Right, see https://docs.temporal.io/dev-guide/java/observability#logging. That can be expanded/improved.

I hadn't seen this before, but it's pretty much what I had in mind when I filed the issue (although I didn't realize how different each SDK was).